llvm/lib/Target/NVPTX/NVPTXInstrInfo.td

```diff
@@ -141,6 +141,7 @@ def hasLDG : Predicate<"Subtarget->hasLDG()">;
 def hasLDU : Predicate<"Subtarget->hasLDU()">;
 def hasPTXASUnreachableBug : Predicate<"Subtarget->hasPTXASUnreachableBug()">;
 def noPTXASUnreachableBug : Predicate<"!Subtarget->hasPTXASUnreachableBug()">;
+def hasO1 : Predicate<"TM.getOptLevel() != CodeGenOptLevel::None">;
```

Comment:
Nit: I'd rename it to something more distinct from `01`. Perhaps `hasOptEnabled`